### PR TITLE
docs: Adds React components naming conventions

### PR DESCRIPTION
**Pull request by Golodhros**
Signed-off-by: Marcos Iglesias <miglesiasvalle@lyft.com>

### Summary of Changes
Adds a naming convention for component names and prop names
_Include a summary of changes then remove this line_

### Documentation
It is documentation only
_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

**Review by ttannis — APPROVED** [review]
Does this mean we plan on doing a mass renaming of things?

File: docs/react_naming_conventions.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+Using this Markdown file:
+
+1. Paste this output into your source file.
+2. See the notes and action items below regarding this conversion run.
+3. Check the rendered output (headings, lists, code blocks, tables) for proper
+   formatting and use a linkchecker before you publish this page.
+
+Conversion notes:
+
+* Docs to Markdown version 1.0β29
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply docs/react_naming_conventions.md:18]: Oopsie! I should remove that! :)

**Comment by Golodhros** [comment]
> Does this mean we plan on doing a mass renaming of things?

Nah, I would say just on the new stuff or when we do some refactors. I don't think the amount of work would be worth it, do you agree?

**Comment by feng-tao** [comment]
> > Does this mean we plan on doing a mass renaming of things?
> 
> Nah, I would say just on the new stuff or when we do some refactors. I don't think the amount of work would be worth it, do you agree?

+1 for new as well as massive refactor may not bring that much benefit in the mean time you loss all the github blame around the code context.

**Pull request merged**